### PR TITLE
The ability passing a named redis cluster connection parameters is added

### PR DESCRIPTION
**Pull request by alexva24**
This ability explained in https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#loading-a-cluster-configuration-by-name

### CODE REVIEW

File: includes/object-cache.php [hunk 1]
```diff
@@ -686,18 +686,22 @@ protected function connect_using_phpredis( $parameters ) {
...
+                    'read_timeout' => $parameters['read_timeout'],
+                    'persistent' => $parameters['persistent'],
+                ];
 
-            if ( isset( $parameters['password'] ) && version_compare( $version, '4.3.0', '>=' ) ) {
-                $args['password'] = $parameters['password'];
-            }
+                if (isset($parameters['password']) && version_compare($version, '4.3.0', '>=')) {
+                    $args['password'] = $parameters['password'];
+                }
```
[the remark this answers is not in the document]
Comment (alexva24) [reply includes/object-cache.php]: Fixed

File: includes/object-cache.php [hunk 2]
```diff
@@ -686,18 +686,22 @@ protected function connect_using_phpredis( $parameters ) {
 
             $this->diagnostics[ 'shards' ] = WP_REDIS_SHARDS;
         } elseif ( defined( 'WP_REDIS_CLUSTER' ) ) {
-            $args = [
-                'cluster' => $this->build_cluster_connection_array(),
-                'timeout' => $parameters['timeout'],
-                'read_timeout' => $parameters['read_timeout'],
-                'persistent' => $parameters['persistent'],
-            ];
+            if ($parameters['cluster_name']) {
```
[the remark this answers is not in the document]
Comment (alexva24) [reply includes/object-cache.php]: I add cluster_name setting. It gives ability define WP_REDIS_CLUSTER_NAME constant.

File: includes/object-cache.php [hunk 3]
```diff
@@ -686,18 +687,22 @@ protected function connect_using_phpredis( $parameters ) {
 
             $this->diagnostics[ 'shards' ] = WP_REDIS_SHARDS;
         } elseif ( defined( 'WP_REDIS_CLUSTER' ) ) {
-            $args = [
-                'cluster' => $this->build_cluster_connection_array(),
-                'timeout' => $parameters['timeout'],
-                'read_timeout' => $parameters['read_timeout'],
-                'persistent' => $parameters['persistent'],
-            ];
+            if ( $parameters['cluster_name'] ) {
```
Comment (tillkruss) [inline includes/object-cache.php:690]: I don't want to introduce *another* constant, please use `WP_REDIS_CLUSTER` and check the type.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: includes/object-cache.php (hunk 3 above)
Comment (alexva24) [reply includes/object-cache.php]: Done

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: includes/object-cache.php [hunk 4]
```diff
@@ -686,18 +686,22 @@ protected function connect_using_phpredis( $parameters ) {
...
+                    'read_timeout' => $parameters['read_timeout'],
+                    'persistent' => $parameters['persistent'],
+                ];
 
-            if ( isset( $parameters['password'] ) && version_compare( $version, '4.3.0', '>=' ) ) {
-                $args['password'] = $parameters['password'];
-            }
+                if ( isset( $parameters['password'] ) && version_compare( $version, '4.3.0', '>=' ) ) {
+                    $args['password'] = $parameters['password'];
+                }
```
[the remark this answers is not in the document]
Comment (alexva24) [reply includes/object-cache.php:701]: There are no tabs in my code. Only spaces

**Pull request merged**